File: java/org.financer.client.javafx/src/main/java/org/financer/client/javafx/util/ApplicationHelper.java
package org.financer.client.javafx.util;

import javafx.application.Platform;
import javafx.application.Preloader;
import javafx.stage.Stage;
import org.financer.client.javafx.main.FinancerApplication;

import java.util.logging.Level;
import java.util.logging.Logger;

public class ApplicationHelper {

    private static final Logger logger = Logger.getLogger("FinancerApplication");

    private ApplicationHelper() {
        super();
    }

    public static void restartApplication(Stage stage) {
        stage.close();
        Platform.runLater(() -> {
            FinancerApplication financerApplication = new FinancerApplication();
            try {
                financerApplication.init();
                financerApplication.start(new Stage());
                financerApplication.notifyPreloader(new Preloader.ProgressNotification(0));
            } catch (Exception e) {
                logger.log(Level.SEVERE, e.getMessage(), e);
            }
        });
    }
}

File: java/org.financer.client.javafx/src/main/kotlin/org/financer/client/javafx/util/ApplicationHelper.kt
package org.financer.client.javafx.util

import javafx.application.Platform
import javafx.application.Preloader.ProgressNotification
import javafx.stage.Stage
import org.financer.client.javafx.main.FinancerApplication
import java.util.logging.Level
import java.util.logging.Logger

object ApplicationHelper {
    private val logger = Logger.getLogger("FinancerApplication")
    @JvmStatic
    fun restartApplication(stage: Stage) {
        stage.close()
        Platform.runLater {
            val financerApplication = FinancerApplication()
            try {
                financerApplication.init()
                financerApplication.start(Stage())
                financerApplication.notifyPreloader(ProgressNotification(0.0))
            } catch (e: Exception) {
                logger.log(Level.SEVERE, e.message, e)
            }
        }
    }
}